client/src/pages/Menu.js

```diff
@@ -14,17 +15,25 @@ const MenuContainer = styled.div`
   }
 `;
 
-const menuOptions = [
-  'Client Master',
-  'ItemMaster',
-  'Storage System',
-  'Inbound',
-  'Outbound',
-];
+const menuRoutes = {
+  'Client Master': 'clientmaster',
+  'Item Master': 'itemmaster',
+  'Storage System': 'storgesystem',
+  Inbound: 'inbound',
+  Outbound: 'outbound',
+};
 
 const CreateMenuOptions = () => {
-  return menuOptions.map((menuOptions) => (
-    <MenuButton key={menuOptions}>{menuOptions}</MenuButton>
+  let history = useHistory();
+
+  function handleOnClick(event) {
+    history.push(menuRoutes[event.currentTarget.value]);
+  }
+
+  return Object.keys(menuRoutes).map((menuOptions) => (
+    <MenuButton key={menuOptions} value={menuOptions} onClick={handleOnClick}>
+      {menuOptions}
+    </MenuButton>
```

Comment:
I see what you want to do here and I like it, but I'd like to propose a different approach. The idea you had before, mapping an array, was a lot more readable than this, simply because of the object `menuRoutes`. I'll add a comment.

`onClick={handleClick}` is best practice

the map is a little confusing because you completely change the parameter name. Disclaimer: if you change this to an array the following suggestion would be obsolete, obviously:

```suggestion
  return Object.keys(menuRoutes).map((menuRoute) => (
    <MenuButton key={menuRoute} value={menuRoute} onClick={handleClick}>
      {menuRoute}
    </MenuButton>
```